src/receive_imf.rs

```diff
@@ -1398,7 +1398,7 @@ async fn lookup_chat_by_reply(
     if let Some(parent) = parent {
         let parent_chat = Chat::load_from_db(context, parent.chat_id).await?;
 
-        if parent.error.is_some() {
+        if !parent.state.is_outgoing() && parent.error.is_some() {
```

Comment:
If the parent message is erroneous it could be assigned to the wrong chat no matter if it was incoming or outgoing, right? Shouldn't we then return `None` in any case?